### PR TITLE
Fixed bug with default parameters NFFT and noverlap in specgram()

### PR DESCRIPTION
**Pull request by vollbier**
While looking at Bug #6064 I stumbled upon the following problem:
The default parameter in `specgram()` of `NFFT` and `noverlap` is ``None``. The fix in #7692 assumes these parameters to be integers. This pull request fixes that.

Also a warning was added to fix Bug #6064.

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -7234,6 +7234,11 @@ def specgram(self, x, NFFT=None, Fs=None, Fc=None, detrend=None,
         Z = np.flipud(Z)
 
         if xextent is None:
+            # define default values:
```
Comment (tacaswell) [inline lib/matplotlib/axes/_axes.py]: Can you move this up to the top?  I think it is better to have this over-ride the defaults if we change the defaults in `mlab.specgram` than to have this silently get out of sync (it is unlikely we will make that change, but this just smell very brittle to me).

File: lib/matplotlib/mlab.py [hunk 2]
```diff
@@ -1274,6 +1274,9 @@ def specgram(x, NFFT=None, Fs=None, detrend=None, window=None,
     """
     if noverlap is None:
         noverlap = 128
+    if len(x) <= (256 if NFFT is None else NFFT):  # see _spectral_helper()
```
Comment (tacaswell) [inline lib/matplotlib/mlab.py]: same here, I would encode the default of 256 so that it can not get out of sync with the down stream functions.

**Comment by tacaswell** [comment]
Tagging this as 2.0 as this fixes a recently merged regression.

**Comment by tacaswell** [comment]
docs failure is expected, #7848 fixes it.

**Comment by tacaswell** [comment]
@efiring I can take care of the backport here

**Comment by tacaswell** [comment]
This makes our tests _super_ noisy :disappointed:

**Comment by efiring** [comment]
Noisy tests, from this?  I don't understand.

**Comment by tacaswell** [comment]
See https://travis-ci.org/matplotlib/matplotlib/jobs/192515102  A lot of the tests of this apparently trigger the warning. 

I am thinking of removing the warning from what in backported and dealing with the warnings just on master.

**Comment by efiring** [comment]
That makes sense to me.

**Comment by tacaswell** [comment]
See #7855

**Pull request merged**